### PR TITLE
Feature/bpn auth validation

### PR DESCRIPTION
**Pull request by andreibogus**
This is implementation of Feature: BPN authorisation validation
Closes: https://github.com/eclipse-tractusx/managed-identity-wallet/issues/278

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [ ] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [ ] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]

File: src/main/java/org/eclipse/tractusx/managedidentitywallets/constant/StringPool.java [hunk 1]
```diff
@@ -102,4 +102,6 @@ private StringPool() {
 
     public static final String PRIVATE_KEY = "PRIVATE KEY";
     public static final String PUBLIC_KEY = "PUBLIC KEY";
+
+    public static final String BPN_NOT_FOUND = "Invalid token, BPN not found";
```
[the remark this answers is not in the document]
Comment (andreibogus) [reply src/main/java/org/eclipse/tractusx/managedidentitywallets/constant/StringPool.java]: Done

**Review by borisrizov-zf — APPROVED** [review]

**Comment by borisrizov-zf** [comment]
@andreibogus @aleksandra-bel please rebase, hopefully DAST will stop whining.

File: src/main/java/org/eclipse/tractusx/managedidentitywallets/config/security/CustomAuthenticationEntryPoint.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                status = ((BearerTokenError) error).getHttpStatus();
+            }
+        }
+
+        if (authException.getMessage().contains(BPN_NOT_FOUND)) {
+            status = HttpStatus.FORBIDDEN;
+        }
+
+        String wwwAuthenticate = computeWWWAuthenticateHeaderValue(parameters);
+        response.addHeader("WWW-Authenticate", wwwAuthenticate);
```
Comment (nitin-vavdiya) [inline src/main/java/org/eclipse/tractusx/managedidentitywallets/config/security/CustomAuthenticationEntryPoint.java:79]: maybe we can use constant from class `org.springframework.http.HttpHeaders`

**Comment by nitin-vavdiya** [comment]
Closing this is PR as there has been no update from the author for a long time.

**Pull request closed without merging**